Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp

```diff
@@ -179,9 +179,11 @@ static constexpr bool verbose = false;
 }
 
 ObjectPropertyCondition generateCondition(
-    VM& vm, JSCell* owner, JSObject* object, UniquedStringImpl* uid, PropertyCondition::Kind conditionKind, Concurrency concurrency)
+    VM& vm, JSCell* owner, JSObject* object, Structure* structure, UniquedStringImpl* uid, PropertyCondition::Kind conditionKind, Concurrency concurrency)
 {
-    Structure* structure = object->structure();
+    if (object->structure() != structure)
+        return ObjectPropertyCondition();
```

Comment:
Let's not add these branches. They're a bit misleading about this structure not changing. I'd vote for a comment or encoding Structure Object pair in a type.